pymc4/inference/sampling.py

```diff
@@ -166,12 +166,11 @@ def build_logp_and_deterministic_functions(
         )
     if state is not None and observed is not None:
         raise ValueError("Can't use both `state` and `observed` arguments")
-    if state is None:
-        state, deterministic_names = initialize_state(model, observed=observed)
-    else:
-        _, st = flow.evaluate_model_transformed(model, state=state)
-        deterministic_names = list(st.deterministics)
-        state = state.as_sampling_state()
+
+    state, deterministic_names = initialize_sampling_state(model, observed=observed, state=state)
+
+    if not state.all_unobserved_values:
+        raise ValueError("The `state` does not contain any `unobserved` values")
```

Comment:
Can this message be a bit more informative? Maybe:

```python
raise ValueError(f"Can not calculate a lot probability: the model {model.name or ""}  has no unobserved values.")
```

Reply:
Ugh. "Can not calculate a *log* probability"

Reply:
Yeah sure, then there is not point to include the additional test case.

Reply:
I thought `with pytest.raises(ValueError)` was a good test case, and confirms the issue was fixed

Reply:
ok, anyway codecov doesn't let me to do it.